### PR TITLE
fix: create in upstream merge

### PR DESCRIPTION
**Pull request by nbaztec**
## Motivation
fixes create

### CODE REVIEW

File: crates/forge/bin/cmd/create.rs [hunk 1]
```diff
@@ -82,11 +95,162 @@ pub struct CreateArgs {
...
+}
+
 impl CreateArgs {
     /// Executes the command to create a contract
     pub async fn run(mut self) -> Result<()> {
         // Find Project & Compile
         let project = self.opts.project()?;
+        let zksync = self.opts.compiler.zk.compile.unwrap_or_default();
+        if zksync {
+            let target_path = if let Some(ref mut path) = self.contract.path {
```
Comment (elfedy) [inline crates/forge/bin/cmd/create.rs:112]: This whole flow + helpers might also fit on a zksync/module crate as a function, that would make it cleaner for merge conflicts as well

**Review by elfedy — APPROVED** [review]

File: crates/script/src/transaction.rs [hunk 2]
```diff
@@ -66,31 +66,6 @@ impl TransactionWithMetadata {
         Self { transaction: WithOtherFields::new(transaction), ..Default::default() }
     }
 
-    pub fn from_zk_tx_request(transaction: TransactionRequest, zk: Option<ZkTransaction>) -> Self {
-        Self { transaction: WithOtherFields::new(transaction), zk, ..Default::default() }
-    }
-
-    pub fn new(
```
Comment (Karrq) [inline crates/script/src/transaction.rs:73]: If we are gonna remove this, doesn't it make more sense to then rename `new_with_zk` to `new`?

**Review by Karrq — APPROVED** [review]
I think we'll need to follow this up with unification of the zk vs evm contracts like we had before (avoiding basically 2 different create depending if we are in zksync mode or not)

File: crates/cli/src/utils/cmd.rs [hunk 3]
```diff
@@ -70,6 +71,35 @@ pub fn remove_contract(
     Ok((abi, bin, runtime))
 }
 
+/// Given a `Project`'s output, removes the matching ABI, Bytecode and
+/// Runtime Bytecode of the given contract.
+#[track_caller]
+pub fn remove_zk_contract(
```
[the remark this answers is not in the document]
Comment (nbaztec) [reply crates/cli/src/utils/cmd.rs:77]: We can refactor this once we have a better picture. 
In general it's easier to have everything in one place, but we need to plan it properly as most times the "zk" functions are a copy-paste of the original ones with a few modifications. So when a upstream merge comes, we want to include those updates to the original function.

File: crates/zksync/core/src/lib.rs [hunk 4]
```diff
@@ -67,43 +73,54 @@ impl ZkTransactionMetadata {
 }
 
 /// Creates a new signed EIP-712 transaction with the provided factory deps.
-pub async fn new_eip712_transaction<M: Middleware, S: Signer>(
-    legacy_or_1559: TypedTransaction,
+pub async fn new_eip712_transaction<
```
[the remark this answers is not in the document]
Comment (nbaztec) [reply crates/zksync/core/src/lib.rs:76]: We can do the refactor once we have a better picture.

**Pull request merged**